File: gntl_app_backend/src/main/java/com/gentlemanstore/cart/dto/CartDTO.java
package com.gentlemanstore.cart.dto;

import lombok.*;

import java.math.BigDecimal;
import java.util.List;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class CartDTO {
    private Long id;
    private List<CartItemDTO> items;
    private BigDecimal totalPrice;
}

File: gntl_app_frontend/app/src/main/java/com/gentlemanstore/feature/cart/data/dto/CartDto.kt
package com.gentlemanstore.feature.cart.data.dto

import java.math.BigDecimal

data class CartItemResponse(
    val id: Long,
    val productName: String,
    val productId: Long,
    val imageUrl: String?,
    val size: String,
    val quantity: Int,
    val price: BigDecimal,
    val originalPrice: BigDecimal?
)

data class CartResponse(
    val id: Long,
    val items: List<CartItemResponse>,
    val totalPrice: BigDecimal
)

data class AddToCartRequest(
    val productId: Long,
    val productSizeId: Long,
    val quantity: Int
)